game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/BattleStep.java

```diff
@@ -28,7 +28,11 @@
     SUB_OFFENSIVE_RETREAT_BEFORE_BATTLE,
     SUB_DEFENSIVE_RETREAT_BEFORE_BATTLE,
     SUBMERGE_SUBS_VS_ONLY_AIR,
+    FIRST_STRIKE_DEFENSIVE_SNEAK_ATTACK,
+    FIRST_STRIKE_OFFENSIVE,
+    FIRST_STRIKE_DEFENSIVE,
```

Comment:
How is FIRST_STRIKE_DEFENSIVE different from FIRST_STRIKE_DEFENSIVE_NO_SNEAK_ATTACK? Perhaps a comment on the enum value would be useful here to describe the intent behind the distinction.

Reply:
I should probably rename the enums.  A defensive sub can fire before the offensive subs fire (FIRST_STRIKE_DEFENSIVE_SNEAK_ATTACK), with the offensive subs (FIRST_STRIKE_DEFENSIVE), or with the other regular units (FIRST_STRIKE_DEFENSIVE_NO_SNEAK_ATTACK).  Any ideas on a name that is easier to understand?  I had thought "FIRST_STRIKE_DEFENSIVE_NO_FIRST_STRIKE" but that felt confusing as well.

Reply:
First, are these enum values persisted as part of `gameData`, are we free to rename the existing ones? 
If not, creating a shim to allow for flexibility perhaps would be good to do sooner before we lock ourselves in.


Perhaps differentiating sneak attack and first strike and make them more mutually exclusive. In this case the order is:
- sneak attack
- first strike
- attack

Reply:
`BattleStep` enums shouldn't be persisted.  They are never used as an object field.

I'm not exactly sure what you mean?  Do you mean to rename the enums as "FIRST_STRIKE_DEFENSIVE_SNEAK_ATTACK", "FIRST_STRIKE_DEFENSIVE_FIRST_STRIKE", and "FIRST_STRIKE_DEFENSIVE_ATTACK"?